Source/WebCore/css/ComputedStyleExtractor.cpp

```diff
@@ -3574,6 +3574,8 @@ RefPtr<CSSValue> ComputedStyleExtractor::valueForPropertyInStyle(const RenderSty
             return CSSPrimitiveValue::create(CSSValueCollapse);
         return CSSPrimitiveValue::create(CSSValueSeparate);
     case CSSPropertyBorderSpacing:
+        if (style.horizontalBorderSpacing() == 0.0 && style.verticalBorderSpacing() == 0.0)
+            return zoomAdjustedPixelValue(style.horizontalBorderSpacing(), style);
         return CSSValueList::createSpaceSeparated(zoomAdjustedPixelValue(style.horizontalBorderSpacing(), style),
             zoomAdjustedPixelValue(style.verticalBorderSpacing(), style));
```

Comment:
The lengths should coalesce in all cases where the horizontal value is equal to the vertical value.

`CSSValuePair::create(zoomAdjustedPixelValue(style.horizontalBorderSpacing(), style), zoomAdjustedPixelValue(style.verticalBorderSpacing(), style));` gives you this functionality for free.

Reply:
(e.g. `1px 1px` should collapse to `1px` as well)

Reply:
You can just replace all those lines to:
```
return CSSValuePair::create(zoomAdjustedPixelValue(style.horizontalBorderSpacing(), style), zoomAdjustedPixelValue(style.verticalBorderSpacing(), style));
```

The serialization of `CSSValuePair` already coalesces identical values (unless you use `CSSValuePair::createNonCoalescing`), see: https://searchfox.org/wubkat/rev/3c50bf12bcc7cc7cb47c8b3a8d238e5f433dcf35/Source/WebCore/css/CSSValuePair.cpp#64-71